### ISSUE TITLE
Move config options from `.pre-commit-config.yaml` to `pyproject.toml`

### ISSUE THREAD

**Issue by mikemhenry** [glotzerlab/signac#418]
When possible, we should put args/options for the tools that we use for pre-commit hooks to be in `pyproject.toml`. That way when people run these tools manually, they pick up the correct options. This also makes matiance easier if we need to make any changes since we only have to look at one file instead of two.

**Comment by bdice** [comment]
@mikemhenry I kept everything that I could in `pyproject.toml` or `setup.cfg`. There are some complications involved. Basically every line in `.pre-commit-config.yaml` falls into one of the following categories. I'll also link a few of the issues that I read while working on #415, for context.

**tl;dr:** I don't think there is any action that can be taken to reduce `.pre-commit-config.yaml` or put its options into `pyproject.toml` or `setup.cfg` -- both of which have to exist.

### Defining repo hooks
```yaml
  - repo: https://...
    rev: 'vX.Y.Z'
    hooks:
      - id: name-of-hook
```
Versions must be specified in `.pre-commit-config.yaml` instead of `requirements-precommit.txt`, if we want to use repo hooks instead of "local" hooks. I think using repo hooks is best, because it prevents users from needing to install the development requirements into their own Python environment. `pre-commit` handles the creation of an isolated virtual environment for each of the repo hooks.

### Exclusions
```yaml
        exclude: '(?:configobj/.*)'
```
`pre-commit` provides _file names_ to the hooks it calls. This means that flake8, for instance, does not apply its exclusions from `setup.cfg`. Thus, handling exclusions for vendored code (`configobj`) must be done in `.pre-commit-config.yaml`:
- https://github.com/pre-commit/pre-commit-hooks/issues/112
- https://gitlab.com/pycqa/flake8/-/issues/10
I also have to exclude `setup.cfg` from the whitespace-altering hooks, but that can't be controlled any other way since it applies to all files in the repo (and thus doesn't read from any Python-specific configuration file).

### Arguments
The only hook taking arguments in `.pre-commit-config.yaml` is `pyupgrade`, and that's because ... it doesn't appear to use `setup.cfg` or `pyproject.toml`. It appears to be strictly designed to be used as a pre-commit hook. 😄

### Consolidating pyproject.toml / setup.cfg
On another note, it is not possible to use only `pyproject.toml` or `setup.cfg` if we rely on both `black` and `flake8`. Both files are necessary:
- https://github.com/psf/black/issues/683
- https://gitlab.com/pycqa/flake8/-/issues/428

**Comment by mikemhenry** [comment]
Okay I remember running into this issue when we first added pre-commit hooks, the only one that really stood out was the args for `pyupgrade` but if it is only used as a pre-commit hook, then that settles that!
